developers/weaviate/current/client-libraries/python.md

```diff
@@ -35,31 +35,70 @@ client.schema.get() # get the full schema as example
 
 ## Authentication
 
-### OIDC authentication flow
-To use the client against an authenticated Weaviate using the OIDC authentication flow, pass the credentials as a parameter when you initialize the client:
+The python client supports multiple OIDC authentication flows to use with Weaviate. 
+
+First, choose one of the supported flows and create the flow-specific authentication configuration, which can then be passed to the client. These secrets are used by the client to obtain an `access token` and (if configured) a `refresh token`.
+
+The `access token` is included in the HTTP-header of any request and is used to authenticate against weaviate. However, this token usually has a limited lifetime and the `refresh token` can be used to acquire a new set of tokens.
+
+### Resource Owner Password flow
+
+This flow authenticates users using their *username* and *password* and should only be used on trusted devices. Both information are not saved in the client and are only used to obtain the first tokens.
+
+Not every provider automatically includes a `refresh token` and an appropriate *scope* might be required that depends on your identity provider. The client uses *offline_access* as default scope which works with some providers.
+
+Without a refresh token, there is no possibility to acquire a new `access token` and the client becomes unauthenticated after expiration.
+
+```python
+import weaviate
+
+resource_owner_config = weaviate.AuthClientPassword(
+  "user", 
```

Comment:
I think for docs it is nicer to explicitly pass the parameter names i.e. `username="user"`

Reply:
good idea, added!